src/ast/desugar.rs

```diff
@@ -689,21 +685,17 @@ impl Desugar {
             .map_err(|e| e.map_token(|tok| tok.to_string()))?)
     }
 
-    pub fn declare(&mut self, name: Symbol, sort: Symbol) -> Vec<NCommand> {
-        let fresh = self.get_fresh();
-        vec![
-            NCommand::Function(FunctionDecl {
-                name: fresh,
-                schema: Schema {
-                    input: vec![],
-                    output: sort,
-                },
-                default: None,
-                merge: None,
-                merge_action: vec![],
-                cost: Some(HIGH_COST),
-            }),
-            NCommand::NormAction(NormAction::Let(name, NormExpr::Call(fresh, vec![]))),
-        ]
+    pub fn declare(&mut self, name: Symbol, sort: Symbol, cost: Option<usize>) -> Vec<NCommand> {
+        vec![NCommand::Function(FunctionDecl {
+            name,
+            schema: Schema {
+                input: vec![],
+                output: sort,
+            },
+            default: None,
+            merge: None,
+            merge_action: vec![],
+            cost,
+        })]
```

Comment:
👍 this seems cleaner than the previous version of creating a randomly named function.